File: usaco/Chapter 1/Section 1.6/numtri/numtri.cpp
/*
ID: deadly51
TASK: numtri
LANG: C++
*/

#include <bits/stdc++.h>

using namespace std;


int main() {
    ifstream fin("numtri.in");
    ofstream fout("numtri.out");
    int n;
    fin >> n;
    vector<vector<int>> piramid(n);
    for (int i = 1; i <= n; i++) {
        piramid[i-1].resize(i);
        for (int j = 0; j < i; j++)
            fin >> piramid[i-1][j];
    }
    int res = 0;
    int min = pow(2, n-1);
    for (int i = min; i < min*2; i++) {
        int sum = 0;
    }
    
    fout << res << "\n";
    return 0;
}

File: usaco/Chapter 1/Section 1.6/numtri/numtri.py
"""
ID: deadly51
TASK: numtri
LANG: PYTHON3
"""
with open("numtri.in", "r") as file:
    n = int(file.readline())
    map = [[int(x) for x in file.readline().split()] for _i in range(n)]

# просто рекурсия с кешом

def wrapper():
    cache = set([])




with open("numtri.out", "w") as file:
    file.write(f"{sum}\n")



